crypto/x509/x509_lu.c

```diff
@@ -93,22 +93,18 @@ int X509_LOOKUP_by_subject_ex(X509_LOOKUP *ctx, X509_LOOKUP_TYPE type,
                               const X509_NAME *name, X509_OBJECT *ret,
                               OSSL_LIB_CTX *libctx, const char *propq)
 {
-    if (ctx->skip
-        || ctx->method == NULL
-        || (ctx->method->get_by_subject == NULL
-            && ctx->method->get_by_subject_ex == NULL))
+    if (ctx->skip || ctx->method == NULL
+           || ctx->method->get_by_subject_ex == NULL)
         return 0;
-    if (ctx->method->get_by_subject_ex != NULL)
-        return ctx->method->get_by_subject_ex(ctx, type, name, ret, libctx,
-                                              propq);
-    else
-        return ctx->method->get_by_subject(ctx, type, name, ret);
+    return ctx->method->get_by_subject_ex(ctx, type, name, ret, libctx, propq);
```

Comment:
This is also a behavior change I am not sure we should do.

Reply:
Yes, this changes the originally implemented behavior because IMHO it was wrong -
see the documentation in `X509_LOOKUP.pod`:
```
X509_LOOKUP_by_subject_ex(), ... 
all return 0 if there is no B<X509_LOOKUP_METHOD> or that
method doesn't implement the corresponding function.
Otherwise, they return what the corresponding function in the
B<X509_LOOKUP_METHOD> returns, which is usually 1 on success and 0 in
error.
```

Reply:
@slontis, since you implemented this in https://github.com/openssl/openssl/pull/12153, what do you think?

Reply:
Ping @slontis - can you please comment?

Reply:
> I don't see a issue with the old code.

Maybe it works fine, but it does not really follow the documentation, as quoted above.
So if we keep the code as it was, please suggest a suitable doc adaptation.

Reply:
Something like the following..

X509_LOOKUP_by_subject_ex() returns 0 if there is no B<X509_LOOKUP_METHOD> that implements either of the get_by_subject_ex() or get_by_subject() functions. If the get_by_subject_ex() is not implemented then it will use the get_by_subject() instead. If the function exists X509_LOOKUP_by_subject_ex() will returns 1 on success and 0 on error.

X509_LOOKUP_by_subject() is similar to X509_LOOKUP_by_subject_ex() but passes NULL for both the libctx and propq.

Reply:
All right, I've corrected the documentation of `X509_LOOKUP_by_subject{,_ex}()` in this spirit
and reverted the proposed changes to their code.